src/cephadm/node-proxy/redfish_dell.py

```diff
@@ -0,0 +1,79 @@
+from redfish_system import RedfishSystem
+from util import logger, normalize_dict
+
+log = logger(__name__)
+
+class RedfishDell(RedfishSystem):
+    def __init__(self, *a, system_endpoint='/Systems/System.Embedded.1', **kw):
+        super().__init__(*a, system_endpoint=system_endpoint, **kw)
+
+    def _update_network(self):
+        net_path = self._system['EthernetInterfaces']['@odata.id']
+        log.info("Updating network")
+        network_info = self.client.get_path(net_path)
+        self._system['network'] = {}
+        result = dict()
+        for interface in network_info['Members']:
+            interface_path = interface['@odata.id']
+            interface_info = self.client.get_path(interface_path)
+            interface_id = interface_info['Id']
+            result[interface_id] = dict()
+            result[interface_id]['description'] = interface_info['Description']
+            result[interface_id]['name'] = interface_info['Name']
+            result[interface_id]['speed_mbps'] = interface_info['SpeedMbps']
+            result[interface_id]['status'] = interface_info['Status']
```

Comment:
I think for this we can just hold a list with the keys that we need to pick from the `interface_info`, something like:

```
fields = ['Description', 'Name', 'SpeedMbps', 'Status']
for f in fields:
     result[interface_id][f] = interface_info[to_snake_case(f)]
```

Where:
```

def to_snake_case(name):
    pattern = re.compile(r'(?<!^)(?=[A-Z])')
    return pattern.sub('_', name).lower()
```
a try-expect would be needed to catch cases where the key doesn't exist on the destination diccionary.

Reply:
good idea, thanks for that suggestion @rkachach